components/shared/net/lib.rs

```diff
@@ -933,6 +933,18 @@ pub enum NetworkError {
     SslValidation(String, Vec<u8>),
     /// Crash error, to be converted to Resource::Crash in the HTML parser.
     Crash(String),
+    UnsupportedScheme,
+    CorsViolation,
+    ConnectionFailure,
+    Timeout,
+    RedirectError,
+    InvalidMethod,
+    ResourceError,
+    SecurityBlock,
```

Comment:
I'm worried that SecurityBlock covers too many different failures modes and will make it harder to track down the cause of errors. Can we make it more granular?